File: TNBSetlistMaker.Domain/Entities/Song.cs
namespace TNBSetlistMaker.Domain.Entities;

public class Song
{
    public Guid Id { get; set; }
    public string SpotifyId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Artist { get; set; } = string.Empty;
    public string? Duration { get; set; } // e.g., "3:45"
    public string? AlbumImageUrl { get; set; }
    public int? Year { get; set; }
    public string? PreviewUrl { get; set; }

    // Relationship: Every song belongs to a playlist
    public Guid PlaylistId { get; set; }
    public Playlist? Playlist { get; set; }
}

File: TNBSetlistMaker.UI/src/types/song.ts
export interface Song {
  id: string;
  spotifyId: string;
  title: string;
  artist: string;
  duration: string | null;
  albumImageUrl: string | null;
  year: number | null;
  playlistId: string;
  playlistName: string;
}

export interface Playlist {
  id: string;
  name: string;
  tag?: string;
}
